docs/src/developer/developer/upgrading.md

```diff
@@ -0,0 +1,32 @@
+# Upgrading
+
+Here are some workflow suggestions when you're trying to upgrade packages (or GHC) in wire-server.
+
+- Use and install https://github.com/maralorn/nix-output-monitor when building nix derivations locally. It shows you the dependency tree building as well as the progress. This can give you a feeling how much progress you've made.
+
+- The `.envrc` environment probably won't load anymore. To fix this adjust the for packages that get passed to `shellFor` in `wire-server.nix`. In the beginning you might want to start out with an empty list as an argument.
+
+- You can explore our nix derivations via `nix repl` and then `:l ./nix` to load the attrset in `./nix/default.nix`. For example to see which version of `aeson` will be used you can evaluate `wireServer.haskellPackages.aeson.version`. TAB-autocompletion also tells you if multiple versions of a package are included in the `nixpkgs` pin, e.g. `aeson_1_5_6_0`, `aeson_2_1_1_0`, which can be used in `manual-overrides.nix` to overwrite the default, e.g. `aeson`.
+
+- Your goal is to make all packages compile again with nix. Start small by trying to build single packages, e.g. `wire-api` or any external dependencies.
+```
+nix-build ./nix -A wireServer.haskellPackagesUnoptimizedNoDocs.wire-api
+```
+
+- When a dependency doesn't build anymore because of unmet version constraints see if you can use a never version (`haskell-pins.nix`, or `manual-overrides.nix` if multiple versions of the package are included in the `nixpkgs` pin). Also check out the git repo or issue tracker of the dependency to find forks that work. You can also try removing versions constraints of packages: see `doJailbreak` in `manual-overrides.nix`.
+
+- If test-suites of a dependency don't compile, or cannot be run because they require IO interaction you can disable them: see `dontCheck` in `manual-overrides.nix`
+
+- To force a rebuild of the nix shell delete the `.env` directory and run `direnv reload`
+
+- If you need to fix code in a dependency consider forking it and creating a PR to the upstream after successful adapation. Clone the repo and then symlink it inside `libs/` and run `generate-local-nix-packages.sh` to temporarily include its dependencies in the development shell. Make sure to include the package in `shellFor`. If you've got a working shell you can check the output of `ghc-pkg dump` to see the nix-provided ghc packages used by cabal.
+
+- Consider using `ghcid --command "cabal repl <package>"` when fixing packages
```

Comment:
Would be nice to give some hints as to why you'd want to do that.

Reply:
You mean you might need to fix code in a dependency? There is only one reason: the dependency is not compiling.

Reply:
Nevermind, I mistead the command :D